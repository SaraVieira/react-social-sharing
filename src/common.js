import styled, { css } from 'styled-components'
import is, { isOr } from 'styled-is'

export const SharingButton = styled.div`
  border-radius: 5px;
  transition: 25ms ease-out;
  padding: 0.5em 0.75em;
  font-family: Helvetica Neue, Helvetica, Arial, sans-serif;
`

export const Link = styled.a`
  text-decoration: none;
  color: #fff;
  margin: 0.5em;
  display: inline-block;

  svg {
    width: 1em;
    height: 1em;
    vertical-align: top;

    &:not(:only-child) {
      margin-right: 0.4em;
    }
  }

  ${is('small')`
    svg {
      vertical-align: middle;
    }
  `};

  ${is('simple')`
    svg {
      stroke: #232323;
      fill: none;
    }
  `};

  ${is('simpleReverse')`
    svg {
      stroke: #fff;
      fill: none;
    }
  `};
`

export const Icon = styled.div`
  display: flex;
  align-items: center;
  fill: #fff;
  stroke: none;

  ${isOr('solid', 'solidcircle')`
    fill: #fff;
    stroke: none
  `};
`
