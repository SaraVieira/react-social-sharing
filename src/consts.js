export default {
  twitter: (link = '', message = '') =>
    `https://twitter.com/intent/tweet/?text=${encodeURIComponent(
      message
    )}&url=${encodeURIComponent(link)}`,
  facebook: (link: '') =>
    `https://facebook.com/sharer/sharer.php?u=${encodeURIComponent(link)}`,
  google: (link: '') =>
    `https://plus.google.com/share?url=${encodeURIComponent(link)}`,
  tumblr: (link: '') =>
    `http://tumblr.com/widgets/share/tool?canonicalUrl=${encodeURIComponent(
      link
    )}`,
  reddit: (link: '') =>
    `https://reddit.com/submit/?url=${encodeURIComponent(link)}`,
  whatsapp: (link = '', message = '') =>
    `https://api.whatsapp.com/send?text=${encodeURIComponent(message)}%20${encodeURIComponent(
      link
    )}`,
  telegram: (link = '', message = '') =>
    `https://telegram.me/share/url?text=${encodeURIComponent(
      message
    )}&url=${encodeURIComponent(link)}`,
  vk: (link = '', message = '') =>
    `http://vk.com/share.php?title=${encodeURIComponent(
      message
    )}&url=${encodeURIComponent(link)}`,
  hacker: (link = '', message = '') =>
    `https://news.ycombinator.com/submitlink?u=${encodeURIComponent(
      link
    )}&t=${encodeURIComponent(message)}`,
  xing: (link = '', message = '') =>
    `https://www.xing.com/app/user?op=share;url=${encodeURIComponent(
      link
    )};title=${encodeURIComponent(message)}.`,
  mail: (link = '', subject, body) =>
    `mailto:?subject=${encodeURIComponent(
      subject || ''
    )}&body=${encodeURIComponent((body && `${body}\n\n${link}`) || link)}`,
  pinterest: (link = '', message = '') =>
    `https://pinterest.com/pin/create/button/?url=${encodeURIComponent(
      link
    )}&media=${encodeURIComponent(link)}&description=${encodeURIComponent(
      message
    )}`,
  linkedin: (link = '', message = '') =>
    `https://www.linkedin.com/shareArticle?mini=true&url=${encodeURIComponent(
      link
    )}&title=${encodeURIComponent(message)}&summary=${encodeURIComponent(
      message
    )}&source=${encodeURIComponent(link)}`
}
