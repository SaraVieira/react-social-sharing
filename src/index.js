// this should be the entry point to your library
module.exports = {
  Twitter: require('./buttons/twitter').default,
  Facebook: require('./buttons/facebook').default,
  Google: require('./buttons/google').default,
  Tumblr: require('./buttons/tumblr').default,
  Mail: require('./buttons/mail').default,
  Pinterest: require('./buttons/pinterest').default,
  Linkedin: require('./buttons/linkedin').default,
  Reddit: require('./buttons/reddit').default,
  Xing: require('./buttons/xing').default,
  Whatsapp: require('./buttons/whatsapp').default,
  HackerNews: require('./buttons/hacker').default,
  VK: require('./buttons/vk').default,
  Telegram: require('./buttons/telegram').default,
  X: require('./buttons/x').default
}
