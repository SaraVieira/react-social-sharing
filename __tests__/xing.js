import React from 'react'
import { Xing } from '../src/index'
import 'jest-styled-components'

describe('Xing', function() {
  test('1', () => {
    const wrapper = mount(
      <Xing
        solidcircle
        big
        message="I am so cool"
        link="http://sharingbuttons.io"
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  test('2', () => {
    const wrapper = mount(
      <Xing
        solidcircle
        medium
        message="I am so cool"
        link="http://sharingbuttons.io"
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
  test('3', () => {
    const wrapper = mount(
      <Xing
        solidcircle
        small
        message="I am so cool"
        link="http://sharingbuttons.io"
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  test('4', () => {
    const wrapper = mount(
      <Xing circle big message="I am so cool" link="http://sharingbuttons.io" />
    )
    expect(wrapper).toMatchSnapshot()
  })
  test('5', () => {
    const wrapper = mount(
      <Xing
        circle
        medium
        message="I am so cool"
        link="http://sharingbuttons.io"
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  test('6', () => {
    const wrapper = mount(
      <Xing
        circle
        small
        message="I am so cool"
        link="http://sharingbuttons.io"
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
  test('7', () => {
    const wrapper = mount(
      <Xing solid big message="I am so cool" link="http://sharingbuttons.io" />
    )
    expect(wrapper).toMatchSnapshot()
  })

  test('8', () => {
    const wrapper = mount(
      <Xing
        solid
        medium
        message="I am so cool"
        link="http://sharingbuttons.io"
      />
    )
    expect(wrapper).toMatchSnapshot()
  })

  test('9', () => {
    const wrapper = mount(
      <Xing
        solid
        small
        message="I am so cool"
        link="http://sharingbuttons.io"
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
  test('11', () => {
    const wrapper = mount(<Xing circle />)
    expect(wrapper).toMatchSnapshot()
  })
  test('11', () => {
    const wrapper = mount(<Xing solidcircle />)
    expect(wrapper).toMatchSnapshot()
  })
  test('12', () => {
    const wrapper = mount(<Xing />)
    expect(wrapper).toMatchSnapshot()
  })
  test('13', () => {
    const wrapper = mount(<Xing simple />)
    expect(wrapper).toMatchSnapshot()
  })
  test('14', () => {
    const wrapper = mount(<Xing simpleReverse />)
    expect(wrapper).toMatchSnapshot()
  })
})
