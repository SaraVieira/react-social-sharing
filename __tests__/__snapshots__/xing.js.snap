// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Xing 1 1`] = `
<Button
  big={true}
  link="http://sharingbuttons.io"
  message="I am so cool"
  solidcircle={true}
>
  <styled.a
    aria-label="Share on Xing"
    href="https://www.xing.com/app/user?op=share;url=http%3A%2F%2Fsharingbuttons.io;title=I%20am%20so%20cool."
    rel="noreferrer noopener"
    target="_blank"
  >
    <a
      aria-label="Share on Xing"
      className="sc-bwzfXH ewcqji"
      href="https://www.xing.com/app/user?op=share;url=http%3A%2F%2Fsharingbuttons.io;title=I%20am%20so%20cool."
      rel="noreferrer noopener"
      target="_blank"
    >
      <styled.div>
        <div
          className="sc-gZMcBi FVSUz"
        >
          <styled.div
            aria-hidden="true"
            solidcircle={true}
          >
            <div
              aria-hidden="true"
              className="sc-htpNat gylFwk"
            >
              <svg
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 0C5.4 0 0 5.4 0 12s5.4 12 12 12 12-5.4 12-12S18.6 0 12 0zM7.8 14.5h-3L7 11.3 5.3 8.5h3l1.8 2.8L8 14.5zm9 5h-3.4l-3-5.5L15 5.5h3.2L13.6 14l3 5.5z"
                />
              </svg>
               
              <span>
                share on Xing
              </span>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </a>
  </styled.a>
</Button>
`;

exports[`Xing 2 1`] = `
<Button
  link="http://sharingbuttons.io"
  medium={true}
  message="I am so cool"
  solidcircle={true}
>
  <styled.a
    aria-label="Share on Xing"
    href="https://www.xing.com/app/user?op=share;url=http%3A%2F%2Fsharingbuttons.io;title=I%20am%20so%20cool."
    rel="noreferrer noopener"
    target="_blank"
  >
    <a
      aria-label="Share on Xing"
      className="sc-bwzfXH ewcqji"
      href="https://www.xing.com/app/user?op=share;url=http%3A%2F%2Fsharingbuttons.io;title=I%20am%20so%20cool."
      rel="noreferrer noopener"
      target="_blank"
    >
      <styled.div>
        <div
          className="sc-gZMcBi FVSUz"
        >
          <styled.div
            aria-hidden="true"
            solidcircle={true}
          >
            <div
              aria-hidden="true"
              className="sc-htpNat gylFwk"
            >
              <svg
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 0C5.4 0 0 5.4 0 12s5.4 12 12 12 12-5.4 12-12S18.6 0 12 0zM7.8 14.5h-3L7 11.3 5.3 8.5h3l1.8 2.8L8 14.5zm9 5h-3.4l-3-5.5L15 5.5h3.2L13.6 14l3 5.5z"
                />
              </svg>
               
              <span>
                Xing
              </span>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </a>
  </styled.a>
</Button>
`;

exports[`Xing 3 1`] = `
<Button
  link="http://sharingbuttons.io"
  message="I am so cool"
  small={true}
  solidcircle={true}
>
  <styled.a
    aria-label="Share on Xing"
    href="https://www.xing.com/app/user?op=share;url=http%3A%2F%2Fsharingbuttons.io;title=I%20am%20so%20cool."
    rel="noreferrer noopener"
    target="_blank"
  >
    <a
      aria-label="Share on Xing"
      className="sc-bwzfXH ewcqji"
      href="https://www.xing.com/app/user?op=share;url=http%3A%2F%2Fsharingbuttons.io;title=I%20am%20so%20cool."
      rel="noreferrer noopener"
      target="_blank"
    >
      <styled.div
        small={true}
      >
        <div
          className="sc-gZMcBi FVSUz"
        >
          <styled.div
            aria-hidden="true"
            solidcircle={true}
          >
            <div
              aria-hidden="true"
              className="sc-htpNat gylFwk"
            >
              <svg
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 0C5.4 0 0 5.4 0 12s5.4 12 12 12 12-5.4 12-12S18.6 0 12 0zM7.8 14.5h-3L7 11.3 5.3 8.5h3l1.8 2.8L8 14.5zm9 5h-3.4l-3-5.5L15 5.5h3.2L13.6 14l3 5.5z"
                />
              </svg>
               
            </div>
          </styled.div>
        </div>
      </styled.div>
    </a>
  </styled.a>
</Button>
`;

exports[`Xing 4 1`] = `
<Button
  big={true}
  circle={true}
  link="http://sharingbuttons.io"
  message="I am so cool"
>
  <styled.a
    aria-label="Share on Xing"
    href="https://www.xing.com/app/user?op=share;url=http%3A%2F%2Fsharingbuttons.io;title=I%20am%20so%20cool."
    rel="noreferrer noopener"
    target="_blank"
  >
    <a
      aria-label="Share on Xing"
      className="sc-bwzfXH ewcqji"
      href="https://www.xing.com/app/user?op=share;url=http%3A%2F%2Fsharingbuttons.io;title=I%20am%20so%20cool."
      rel="noreferrer noopener"
      target="_blank"
    >
      <styled.div>
        <div
          className="sc-gZMcBi FVSUz"
        >
          <styled.div
            aria-hidden="true"
          >
            <div
              aria-hidden="true"
              className="sc-htpNat dZQCuR"
            >
              <svg
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="11.5"
                />
                <path
                  d="M8.4 8.5h-3L7 11.3l-2.2 3.2h3l2.3-3.2zm10-3h-3.2l-5 8.5 3.2 5.5h3.3l-3-5.5z"
                />
              </svg>
               
              <span>
                share on Xing
              </span>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </a>
  </styled.a>
</Button>
`;

exports[`Xing 5 1`] = `
<Button
  circle={true}
  link="http://sharingbuttons.io"
  medium={true}
  message="I am so cool"
>
  <styled.a
    aria-label="Share on Xing"
    href="https://www.xing.com/app/user?op=share;url=http%3A%2F%2Fsharingbuttons.io;title=I%20am%20so%20cool."
    rel="noreferrer noopener"
    target="_blank"
  >
    <a
      aria-label="Share on Xing"
      className="sc-bwzfXH ewcqji"
      href="https://www.xing.com/app/user?op=share;url=http%3A%2F%2Fsharingbuttons.io;title=I%20am%20so%20cool."
      rel="noreferrer noopener"
      target="_blank"
    >
      <styled.div>
        <div
          className="sc-gZMcBi FVSUz"
        >
          <styled.div
            aria-hidden="true"
          >
            <div
              aria-hidden="true"
              className="sc-htpNat dZQCuR"
            >
              <svg
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="11.5"
                />
                <path
                  d="M8.4 8.5h-3L7 11.3l-2.2 3.2h3l2.3-3.2zm10-3h-3.2l-5 8.5 3.2 5.5h3.3l-3-5.5z"
                />
              </svg>
               
              <span>
                Xing
              </span>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </a>
  </styled.a>
</Button>
`;

exports[`Xing 6 1`] = `
<Button
  circle={true}
  link="http://sharingbuttons.io"
  message="I am so cool"
  small={true}
>
  <styled.a
    aria-label="Share on Xing"
    href="https://www.xing.com/app/user?op=share;url=http%3A%2F%2Fsharingbuttons.io;title=I%20am%20so%20cool."
    rel="noreferrer noopener"
    target="_blank"
  >
    <a
      aria-label="Share on Xing"
      className="sc-bwzfXH ewcqji"
      href="https://www.xing.com/app/user?op=share;url=http%3A%2F%2Fsharingbuttons.io;title=I%20am%20so%20cool."
      rel="noreferrer noopener"
      target="_blank"
    >
      <styled.div
        small={true}
      >
        <div
          className="sc-gZMcBi FVSUz"
        >
          <styled.div
            aria-hidden="true"
          >
            <div
              aria-hidden="true"
              className="sc-htpNat dZQCuR"
            >
              <svg
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="11.5"
                />
                <path
                  d="M8.4 8.5h-3L7 11.3l-2.2 3.2h3l2.3-3.2zm10-3h-3.2l-5 8.5 3.2 5.5h3.3l-3-5.5z"
                />
              </svg>
               
            </div>
          </styled.div>
        </div>
      </styled.div>
    </a>
  </styled.a>
</Button>
`;

exports[`Xing 7 1`] = `
<Button
  big={true}
  link="http://sharingbuttons.io"
  message="I am so cool"
  solid={true}
>
  <styled.a
    aria-label="Share on Xing"
    href="https://www.xing.com/app/user?op=share;url=http%3A%2F%2Fsharingbuttons.io;title=I%20am%20so%20cool."
    rel="noreferrer noopener"
    target="_blank"
  >
    <a
      aria-label="Share on Xing"
      className="sc-bwzfXH ewcqji"
      href="https://www.xing.com/app/user?op=share;url=http%3A%2F%2Fsharingbuttons.io;title=I%20am%20so%20cool."
      rel="noreferrer noopener"
      target="_blank"
    >
      <styled.div>
        <div
          className="sc-gZMcBi FVSUz"
        >
          <styled.div
            aria-hidden="true"
            solid={true}
          >
            <div
              aria-hidden="true"
              className="sc-htpNat gylFwk"
            >
              <svg
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M10.2 9.7l-3-5.4C7.2 4 7 4 6.8 4h-5c-.3 0-.4 0-.5.2v.5L4 10 .4 16v.5c0 .2.2.3.4.3h5c.3 0 .4 0 .5-.2l4-6.6v-.5zM24 .2l-.5-.2H18s-.2 0-.3.3l-8 14v.4l5.2 9c0 .2 0 .3.3.3h5.4s.3 0 .4-.2c.2-.2.2-.4 0-.5l-5-8.8L24 .7V.2z"
                />
              </svg>
               
              <span>
                share on Xing
              </span>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </a>
  </styled.a>
</Button>
`;

exports[`Xing 8 1`] = `
<Button
  link="http://sharingbuttons.io"
  medium={true}
  message="I am so cool"
  solid={true}
>
  <styled.a
    aria-label="Share on Xing"
    href="https://www.xing.com/app/user?op=share;url=http%3A%2F%2Fsharingbuttons.io;title=I%20am%20so%20cool."
    rel="noreferrer noopener"
    target="_blank"
  >
    <a
      aria-label="Share on Xing"
      className="sc-bwzfXH ewcqji"
      href="https://www.xing.com/app/user?op=share;url=http%3A%2F%2Fsharingbuttons.io;title=I%20am%20so%20cool."
      rel="noreferrer noopener"
      target="_blank"
    >
      <styled.div>
        <div
          className="sc-gZMcBi FVSUz"
        >
          <styled.div
            aria-hidden="true"
            solid={true}
          >
            <div
              aria-hidden="true"
              className="sc-htpNat gylFwk"
            >
              <svg
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M10.2 9.7l-3-5.4C7.2 4 7 4 6.8 4h-5c-.3 0-.4 0-.5.2v.5L4 10 .4 16v.5c0 .2.2.3.4.3h5c.3 0 .4 0 .5-.2l4-6.6v-.5zM24 .2l-.5-.2H18s-.2 0-.3.3l-8 14v.4l5.2 9c0 .2 0 .3.3.3h5.4s.3 0 .4-.2c.2-.2.2-.4 0-.5l-5-8.8L24 .7V.2z"
                />
              </svg>
               
              <span>
                Xing
              </span>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </a>
  </styled.a>
</Button>
`;

exports[`Xing 9 1`] = `
<Button
  link="http://sharingbuttons.io"
  message="I am so cool"
  small={true}
  solid={true}
>
  <styled.a
    aria-label="Share on Xing"
    href="https://www.xing.com/app/user?op=share;url=http%3A%2F%2Fsharingbuttons.io;title=I%20am%20so%20cool."
    rel="noreferrer noopener"
    target="_blank"
  >
    <a
      aria-label="Share on Xing"
      className="sc-bwzfXH ewcqji"
      href="https://www.xing.com/app/user?op=share;url=http%3A%2F%2Fsharingbuttons.io;title=I%20am%20so%20cool."
      rel="noreferrer noopener"
      target="_blank"
    >
      <styled.div
        small={true}
      >
        <div
          className="sc-gZMcBi FVSUz"
        >
          <styled.div
            aria-hidden="true"
            solid={true}
          >
            <div
              aria-hidden="true"
              className="sc-htpNat gylFwk"
            >
              <svg
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M10.2 9.7l-3-5.4C7.2 4 7 4 6.8 4h-5c-.3 0-.4 0-.5.2v.5L4 10 .4 16v.5c0 .2.2.3.4.3h5c.3 0 .4 0 .5-.2l4-6.6v-.5zM24 .2l-.5-.2H18s-.2 0-.3.3l-8 14v.4l5.2 9c0 .2 0 .3.3.3h5.4s.3 0 .4-.2c.2-.2.2-.4 0-.5l-5-8.8L24 .7V.2z"
                />
              </svg>
               
            </div>
          </styled.div>
        </div>
      </styled.div>
    </a>
  </styled.a>
</Button>
`;

exports[`Xing 11 1`] = `
<Button
  circle={true}
>
  <styled.a
    aria-label="Share on Xing"
    href="https://www.xing.com/app/user?op=share;url=;title=undefined."
    rel="noreferrer noopener"
    target="_blank"
  >
    <a
      aria-label="Share on Xing"
      className="sc-bwzfXH ewcqji"
      href="https://www.xing.com/app/user?op=share;url=;title=undefined."
      rel="noreferrer noopener"
      target="_blank"
    >
      <styled.div>
        <div
          className="sc-gZMcBi FVSUz"
        >
          <styled.div
            aria-hidden="true"
          >
            <div
              aria-hidden="true"
              className="sc-htpNat dZQCuR"
            >
              <svg
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <circle
                  cx="12"
                  cy="12"
                  r="11.5"
                />
                <path
                  d="M8.4 8.5h-3L7 11.3l-2.2 3.2h3l2.3-3.2zm10-3h-3.2l-5 8.5 3.2 5.5h3.3l-3-5.5z"
                />
              </svg>
               
              <span>
                Xing
              </span>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </a>
  </styled.a>
</Button>
`;

exports[`Xing 11 2`] = `
<Button
  solidcircle={true}
>
  <styled.a
    aria-label="Share on Xing"
    href="https://www.xing.com/app/user?op=share;url=;title=undefined."
    rel="noreferrer noopener"
    target="_blank"
  >
    <a
      aria-label="Share on Xing"
      className="sc-bwzfXH ewcqji"
      href="https://www.xing.com/app/user?op=share;url=;title=undefined."
      rel="noreferrer noopener"
      target="_blank"
    >
      <styled.div>
        <div
          className="sc-gZMcBi FVSUz"
        >
          <styled.div
            aria-hidden="true"
            solidcircle={true}
          >
            <div
              aria-hidden="true"
              className="sc-htpNat gylFwk"
            >
              <svg
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M12 0C5.4 0 0 5.4 0 12s5.4 12 12 12 12-5.4 12-12S18.6 0 12 0zM7.8 14.5h-3L7 11.3 5.3 8.5h3l1.8 2.8L8 14.5zm9 5h-3.4l-3-5.5L15 5.5h3.2L13.6 14l3 5.5z"
                />
              </svg>
               
              <span>
                Xing
              </span>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </a>
  </styled.a>
</Button>
`;

exports[`Xing 12 1`] = `
<Button>
  <styled.a
    aria-label="Share on Xing"
    href="https://www.xing.com/app/user?op=share;url=;title=undefined."
    rel="noreferrer noopener"
    target="_blank"
  >
    <a
      aria-label="Share on Xing"
      className="sc-bwzfXH ewcqji"
      href="https://www.xing.com/app/user?op=share;url=;title=undefined."
      rel="noreferrer noopener"
      target="_blank"
    >
      <styled.div>
        <div
          className="sc-gZMcBi FVSUz"
        >
          <styled.div
            aria-hidden="true"
            solid={true}
          >
            <div
              aria-hidden="true"
              className="sc-htpNat gylFwk"
            >
              <svg
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M10.2 9.7l-3-5.4C7.2 4 7 4 6.8 4h-5c-.3 0-.4 0-.5.2v.5L4 10 .4 16v.5c0 .2.2.3.4.3h5c.3 0 .4 0 .5-.2l4-6.6v-.5zM24 .2l-.5-.2H18s-.2 0-.3.3l-8 14v.4l5.2 9c0 .2 0 .3.3.3h5.4s.3 0 .4-.2c.2-.2.2-.4 0-.5l-5-8.8L24 .7V.2z"
                />
              </svg>
               
              <span>
                Xing
              </span>
            </div>
          </styled.div>
        </div>
      </styled.div>
    </a>
  </styled.a>
</Button>
`;

exports[`Xing 13 1`] = `
<Button
  simple={true}
>
  <styled.a
    aria-label="Share on Xing"
    href="https://www.xing.com/app/user?op=share;url=;title=undefined."
    rel="noreferrer noopener"
    simple={true}
    target="_blank"
  >
    <a
      aria-label="Share on Xing"
      className="sc-bwzfXH ljNxya"
      href="https://www.xing.com/app/user?op=share;url=;title=undefined."
      rel="noreferrer noopener"
      target="_blank"
    >
      <XingIconFill>
        <svg
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10.2 9.7l-3-5.4C7.2 4 7 4 6.8 4h-5c-.3 0-.4 0-.5.2v.5L4 10 .4 16v.5c0 .2.2.3.4.3h5c.3 0 .4 0 .5-.2l4-6.6v-.5zM24 .2l-.5-.2H18s-.2 0-.3.3l-8 14v.4l5.2 9c0 .2 0 .3.3.3h5.4s.3 0 .4-.2c.2-.2.2-.4 0-.5l-5-8.8L24 .7V.2z"
          />
        </svg>
      </XingIconFill>
    </a>
  </styled.a>
</Button>
`;

exports[`Xing 14 1`] = `
<Button
  simpleReverse={true}
>
  <styled.a
    aria-label="Share on Xing"
    href="https://www.xing.com/app/user?op=share;url=;title=undefined."
    rel="noreferrer noopener"
    simpleReverse={true}
    target="_blank"
  >
    <a
      aria-label="Share on Xing"
      className="sc-bwzfXH dqksVq"
      href="https://www.xing.com/app/user?op=share;url=;title=undefined."
      rel="noreferrer noopener"
      target="_blank"
    >
      <XingIconFill>
        <svg
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M10.2 9.7l-3-5.4C7.2 4 7 4 6.8 4h-5c-.3 0-.4 0-.5.2v.5L4 10 .4 16v.5c0 .2.2.3.4.3h5c.3 0 .4 0 .5-.2l4-6.6v-.5zM24 .2l-.5-.2H18s-.2 0-.3.3l-8 14v.4l5.2 9c0 .2 0 .3.3.3h5.4s.3 0 .4-.2c.2-.2.2-.4 0-.5l-5-8.8L24 .7V.2z"
          />
        </svg>
      </XingIconFill>
    </a>
  </styled.a>
</Button>
`;
